### PR TITLE
OADP-4160-Added dpa config for burst and qps

### PR DESCRIPTION
**Pull request by shdeshpa07**
## Jira 

* [OADP-4160](https://issues.redhat.com/browse/OADP-4160)

Added a section for DPA configuration for burst and qps.

##  Version

* OCP 4.13 → OCP 4.18

## Preview

*  [AWS DPA](https://82757--ocpdocs-pr.netlify.app/openshift-enterprise/latest/backup_and_restore/application_backup_and_restore/installing/installing-oadp-aws.html#oadp-configuring-client-burst-qps_installing-oadp-aws)
* [GCP DPA](https://82757--ocpdocs-pr.netlify.app/openshift-enterprise/latest/backup_and_restore/application_backup_and_restore/installing/installing-oadp-gcp#oadp-configuring-client-burst-qps_installing-oadp-gcp)
* [Azure DPA](https://82757--ocpdocs-pr.netlify.app/openshift-enterprise/latest/backup_and_restore/application_backup_and_restore/installing/installing-oadp-azure#oadp-configuring-client-burst-qps_installing-oadp-azure)
* [MCG DPA](https://82757--ocpdocs-pr.netlify.app/openshift-enterprise/latest/backup_and_restore/application_backup_and_restore/installing/installing-oadp-mcg#oadp-configuring-client-burst-qps_installing-oadp-mcg)
* [ODF DPA](https://82757--ocpdocs-pr.netlify.app/openshift-enterprise/latest/backup_and_restore/application_backup_and_restore/installing/installing-oadp-ocs#oadp-configuring-client-burst-qps_installing-oadp-ocs)
* [Virt DPA](https://82757--ocpdocs-pr.netlify.app/openshift-enterprise/latest/backup_and_restore/application_backup_and_restore/installing/installing-oadp-kubevirt#oadp-configuring-client-burst-qps_installing-oadp-kubevirt)

## QE Review

* [ ] QE has approved this change.

### CODE REVIEW

**Comment by shdeshpa07** [comment]
@anarnold97 - PR is ready for your review. Could you please review. Thanks.

File: modules/oadp-configuring-client-burst-qps.adoc [hunk 1]
```diff
@@ -0,0 +1,59 @@
+// Module included in the following assemblies:
+//
+// * backup_and_restore/application_backup_and_restore/installing/installing-oadp-aws.adoc
+
+
+:_mod-docs-content-type: PROCEDURE
+[id="oadp-configuring-client-burst-qps_{context}"]
+= Configuring the DPA with client burst and queries per second (QPS) settings
+
+You can set the burst and QPS values of the `velero` server by configuring the Data Protection Application (DPA) with the burst and QPS values. You can use the `dpa.configuration.velero.client-burst` and `dpa.configuration.velero.client-qps` fields of the DPA to set the burst and QPS values.
```
Comment (anarnold97) [inline modules/oadp-configuring-client-burst-qps.adoc:10]: Do you think we should add a QPS definition to an introductory paragraph, something like

```
The queries per second (QPS) setting determines the number of client requests or queries that are sent to the Velero server. While the burst value determines how many requests from the client can be stored for processing.
```

Then we could shorten the title to 

```
= Configuring the DPA with client burst and QPS settings
```

Comment (shdeshpa07) [reply modules/oadp-configuring-client-burst-qps.adoc:12]: Done. thank you.

**Comment by shdeshpa07** [comment]
@weshayutin  @amastbau  - Could you please review this PR? Thanks.

File: modules/oadp-configuring-client-burst-qps.adoc [hunk 2]
```diff
@@ -0,0 +1,62 @@
+// Module included in the following assemblies:
+//
+// * backup_and_restore/application_backup_and_restore/installing/installing-oadp-aws.adoc
+
+
+:_mod-docs-content-type: PROCEDURE
+[id="oadp-configuring-client-burst-qps_{context}"]
+= Configuring the DPA with client burst and QPS settings
+
+The queries per second (QPS) setting determines the number of client requests or queries that are sent to the `velero` server. The burst setting determines how many requests from the client can be stored for processing.
```
Comment (sseago) [inline modules/oadp-configuring-client-burst-qps.adoc:10]: The burst setting determines how many requests will be sent before limits will be applied. Once the burst limit is hit, QPS determines how many additional requests will be sent per second.

File: modules/oadp-configuring-client-burst-qps.adoc [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+          bucket: <bucket_name> 
+          prefix: velero
+        provider: aws
+  configuration:
+    nodeAgent:
+      enable: true
+      uploaderType: restic
+    velero:
+      client-burst: 300 # <1>
+      client-qps: 500 # <2>
```
Comment (sseago) [inline modules/oadp-configuring-client-burst-qps.adoc:55]: I would update the example to have these either the same or have burst larger than qps. Maybe just reverse the two numbers here.

**Review by sseago — CHANGES_REQUESTED** [review]

File: modules/oadp-configuring-client-burst-qps.adoc (hunk 2 above)
Comment (shdeshpa07) [reply modules/oadp-configuring-client-burst-qps.adoc]: Done. thanks.

File: modules/oadp-configuring-client-burst-qps.adoc (hunk 3 above)
Comment (shdeshpa07) [reply modules/oadp-configuring-client-burst-qps.adoc]: Done. thanks.

File: modules/oadp-configuring-client-burst-qps.adoc [hunk 4]
```diff
@@ -0,0 +1,62 @@
+// Module included in the following assemblies:
+//
+// * backup_and_restore/application_backup_and_restore/installing/installing-oadp-aws.adoc
+
+
+:_mod-docs-content-type: PROCEDURE
+[id="oadp-configuring-client-burst-qps_{context}"]
+= Configuring the DPA with client burst and QPS settings
+
+The burst setting determines how many requests can be sent to the `velero` server, before the limit is applied. After the burst limit is reached, the queries per second (QPS) setting determines how many additional requests can be sent, per second.
```
Comment (adellape) [inline modules/oadp-configuring-client-burst-qps.adoc:10]: Maybe drop both commas:
```suggestion
The burst setting determines how many requests can be sent to the `velero` server before the limit is applied. After the burst limit is reached, the queries per second (QPS) setting determines how many additional requests can be sent per second.
```

**Comment by adellape** [comment]
@shdeshpa07 Should it also be 4.18?

**Comment by shdeshpa07** [comment]
> @shdeshpa07 Should it also be 4.18?

Yes @adellape , you are right. It should also be 4.18. Thank you for that.

**Review by sseago — APPROVED** [review]

**Pull request merged**